core/shared/src/main/scala/org/http4s/EntityEncoder.scala

```diff
@@ -101,14 +101,12 @@ object EntityEncoder {
     *
     * This constructor is a helper for types that can be serialized synchronously, for example a String.
     */
-  def simple[A](hs: Header.ToRaw*)(toChunk: A => Chunk[Byte]): EntityEncoder.Pure[A] =
-    encodeBy(hs: _*)(a => Entity.strict(toChunk(a)))
+  def simple[A](hs: Header.ToRaw*)(toByteVector: A => ByteVector): EntityEncoder.Pure[A] =
+    encodeBy(hs: _*)(a => Entity.strict(toByteVector(a)))
 
   /** Encodes a value from its Show instance.  Too broad to be implicit, too useful to not exist. */
-  def showEncoder[A](implicit charset: Charset = `UTF-8`, show: Show[A]): EntityEncoder.Pure[A] = {
-    val hdr = `Content-Type`(MediaType.text.plain).withCharset(charset)
-    simple[A](hdr)(a => Chunk.array(show.show(a).getBytes(charset.nioCharset)))
-  }
+  def showEncoder[A](implicit charset: Charset = `UTF-8`, show: Show[A]): EntityEncoder.Pure[A] =
```

Comment:
This is irrelevant to current PR, but having `Charset` in implicit scope isn't that a good design, I guess. It isn't a typeclass. Should we discuss this before we release 1.0?

Reply:
Yeah definitely worth a discussion. I agree that anything `implicit`-based should be well-justified.